src/bin/pgcopydb/pgsql.h

```diff
@@ -337,9 +337,16 @@ typedef struct CopyArgs
 	bool truncate;
 	bool freeze;
 	uint64_t bytesTransmitted;
+	uint64_t bytesTransmittedBeforeSavingProgress; /* the bytes transmitted before saving the progress to DB */
+	uint64_t lastSavingTimeMs; /* the last saving time of the progress to the DB */
```

Comment:
That's a callback responsibility, please consider moving that away from here and within the `CopyProgressContext` structure, where it belongs.